Description: The patch fixes mysqld crash on shutdown when built
with CFLAGS="-g -Wl,-Bsymbolic-functions" (which are enabled
by default on Ubuntu when building packages).

Remove this once upstream has released an official fix.
Author: Alexander Barkov <bar@mariadb.org>
Bug: https://mariadb.atlassian.net/browse/MDEV-9091


--- a/storage/tokudb/ft-index/portability/CMakeLists.txt
+++ b/storage/tokudb/ft-index/portability/CMakeLists.txt
@@ -14,12 +14,12 @@ set(tokuportability_srcs
   )
 
 add_library(${LIBTOKUPORTABILITY} SHARED ${tokuportability_srcs})
-target_link_libraries(${LIBTOKUPORTABILITY} LINK_PRIVATE ${LIBJEMALLOC})
+target_link_libraries(${LIBTOKUPORTABILITY})
 target_link_libraries(${LIBTOKUPORTABILITY} LINK_PUBLIC ${CMAKE_THREAD_LIBS_INIT} ${EXTRA_SYSTEM_LIBS})
 
 add_library(tokuportability_static_conv STATIC ${tokuportability_srcs})
 set_target_properties(tokuportability_static_conv PROPERTIES POSITION_INDEPENDENT_CODE ON)
-set(tokuportability_source_libs tokuportability_static_conv ${LIBJEMALLOC} ${CMAKE_THREAD_LIBS_INIT} ${EXTRA_SYSTEM_LIBS})
+set(tokuportability_source_libs tokuportability_static_conv ${CMAKE_THREAD_LIBS_INIT} ${EXTRA_SYSTEM_LIBS})
 toku_merge_static_libs(${LIBTOKUPORTABILITY}_static ${LIBTOKUPORTABILITY}_static "${tokuportability_source_libs}")
 
 maybe_add_gcov_to_libraries(${LIBTOKUPORTABILITY} tokuportability_static_conv)
